datalad/metadata/aggregate.py

```diff
@@ -0,0 +1,124 @@
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the datalad package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""Interface for managing metadata
+"""
+
+__docformat__ = 'restructuredtext'
+
+from simplejson import dump as jsondump
+import os
+from os.path import join as opj, exists
+from datalad.interface.base import Interface
+from datalad.interface.common_opts import recursion_limit, recursion_flag
+from datalad.distribution.dataset import datasetmethod, EnsureDataset, \
+    Dataset, require_dataset
+from ..support.param import Parameter
+from ..support.constraints import EnsureNone
+from ..log import lgr
+from . import get_metadata, extract_metadata, metadata_filename, metadata_basepath, \
+    json_dump_kwargs, flatten_metadata_graph
+
+
+def _optimize_jsonld(obj):
+    try:
+        from pyld import jsonld
+    except ImportError:
+        lgr.debug('pyld not available, not compacting meta data graph')
+        return obj
+    return flatten_metadata_graph(obj)
+
+
+def _store_json(path, meta, optimize=True):
+
+    if not exists(path):
+        os.makedirs(path)
+
+    fname = opj(path, metadata_filename)
+
+    if optimize:
+        meta = _optimize_jsonld(meta)
```

Comment:
Not sure,if we want to keep this optimization here at all. Will now when query is implemented. Until then, no test.